link.go

```diff
@@ -758,19 +757,6 @@ func (l *link) muxHandleFrame(fr frames.FrameBody) error {
 	return nil
 }
 
```

Comment:
The Check() call is racy though, i.e. the link might be alive when you call it but detach/close right after.  I think the only safe way to handle this is when selecting on channels to include l.Detached (like how we do for conn.Done etc).

Reply:
If there's a real need to know outside of go-amqp when a link has been detached we can add something like a `Detached()` method on `Sender` and `Receiver` so that callers can react to this event.

Reply:
Opened https://github.com/Azure/go-amqp/issues/99 to track adding `Detached()`